src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,82 @@
+package duke;
+
+import duke.command.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Parses user input to create the appropriate command for Duke.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input and returns the corresponding Duke command.
+     *
+     * @param input The user input to be parsed.
+     * @return A Command object representing the user's request.
+     * @throws DukeException If the input is invalid or contains errors.
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] parts = input.split(" ", 2);
+        String commandType = parts[0];
+
+        switch (commandType) {
```

Comment:
I like the use of the various Command Classes to encapsulate the handling of each command